packages/node_modules/function-tree/src/tags/index.js

```diff
@@ -0,0 +1,46 @@
+import Tag from './Tag'
+
+export function convertObjectWithTemplates(obj, resolve) {
+  if (resolve.isTag(obj)) {
+    return resolve.value(obj)
+  }
+
+  return Object.keys(obj).reduce((convertedObject, key) => {
+    convertedObject[key] = resolve.value(obj[key])
+
+    return convertedObject
+  }, {})
+}
+
+export function extractValueWithPath(obj, path) {
+  return path.split('.').reduce((currentValue, key, index) => {
+    if (index > 0 && currentValue === undefined) {
+      throw new Error(
+        `A tag is extracting with path "${path}", but it is not valid`
```

Comment:
Error messages are phrases. Needs a `.` at the end (same for other error messages below)...